Typing errors

```
src/game.py:4: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/game.py:5: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/game.py:6: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/game.py:7: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/game.py:8: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/game.py:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/settings.py:9: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/spoiler_file.py:7: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/spoiler_file.py:8: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
src/spoiler_file.py:15: error: Incompatible return value type (got "SpoilerStatusEnum", expected "int")  [return-value]
src/spoiler_file.py:18: error: Incompatible return value type (got "SpoilerStatusEnum", expected "int")  [return-value]
src/gui/notification_dialog.py:18: error: "type[QDialogButtonBox]" has no attribute "Close"  [attr-defined]
src/gui/game_layout.py:37: error: Incompatible types in assignment (expression has type "QGridLayout", base class "QWidget" defined the type as "Callable[[QWidget], QLayout]")  [assignment]
src/gui/game_layout.py:76: error: "type[Qt]" has no attribute "AlignCenter"  [attr-defined]
src/gui/main_window.py:26: error: "type[Qt]" has no attribute "ScrollBarAsNeeded"  [attr-defined]
src/gui/main_window.py:27: error: "type[Qt]" has no attribute "ScrollBarAsNeeded"  [attr-defined]
src/gui/main_window.py:59: error: Argument 1 to "load_file" of "MainWindow" has incompatible type "str | None"; expected "str"  [arg-type]
src/gui/main_window.py:151: error: Function "hasUrls" could always be true in boolean context  [truthy-function]
src/gui/main_window.py:152: error: "type[Qt]" has no attribute "CopyAction"  [attr-defined]
src/gui/main_window.py:159: error: Function "hasUrls" could always be true in boolean context  [truthy-function]
src/gui/main_window.py:160: error: "type[Qt]" has no attribute "CopyAction"  [attr-defined]
Found 12 errors in 4 files (checked 16 source files)

```

As of 20e1b97f45154f81c9d5107322ec295b311c9371

--- a/src/arg_parser.py
+++ b/src/arg_parser.py
@@ -2,6 +2,9 @@ import argparse
 
 
 class ArgParser:
+    parser: argparse.ArgumentParser
+    args: argparse.Namespace
+
     def __init__(self):
         self.parser = self.define_args()
         self.args = self.parser.parse_args()

--- a/src/game.py
+++ b/src/game.py
@@ -1,12 +1,20 @@
 class Game:
+    dock_connections: list
+    has_hints: bool
+    major_items: list
+    minor_items: list
+    nothing_item: str
+    special_configurations: list
+    victory_key: str
+
     def __init__(self):
         self.dock_connections = None  # Meant for elevator rando; currently unsupported
-        self.has_hints: bool = False
-        self.major_items: list = []
-        self.minor_items: list = []
-        self.nothing_item: str = "Energy Transfer Module"
-        self.special_configurations: list = []
-        self.victory_key: str = "A default victory key"
+        self.has_hints = False
+        self.major_items = []
+        self.minor_items = []
+        self.nothing_item = "Energy Transfer Module"
+        self.special_configurations = []
+        self.victory_key = "A default victory key"
 
 
 class NotSupportedGame(Game):

--- a/src/gui/game_layout.py
+++ b/src/gui/game_layout.py
@@ -1,3 +1,4 @@
+from typing import Callable
 from PySide6 import QtCore, QtWidgets
 from world import World
 from gui.notification_dialog import NotificationDialog
@@ -11,10 +12,15 @@ from gui.layouts.fusion_style import FusionStyle
 
 
 class GameLayout(QtWidgets.QWidget):
+    game_style: BaseLayout
+    world: World
+    # We override this with the layout we intend to use
+    layout: Callable[..., QtWidgets.QGridLayout]
+
     def __init__(self, world: World):
         super().__init__()
-        self.game_style: BaseLayout = BaseLayout()
-        self.world: World = world
+        self.game_style = BaseLayout()
+        self.world = world
         unsupported_game = False
 
         match self.world.game_id:
@@ -35,15 +41,15 @@ class GameLayout(QtWidgets.QWidget):
             NotificationDialog.show(self, "Error", str(e))
             return
 
-        self.layout: QtWidgets.QGridLayout = QtWidgets.QGridLayout(self)
-        self.layout.setColumnStretch(0, 20)
-        self.layout.setColumnStretch(1, 30)
-        self.layout.setColumnStretch(2, 50)
+        self.setLayout(QtWidgets.QGridLayout(self))
+        self.layout().setColumnStretch(0, 20)
+        self.layout().setColumnStretch(1, 30)
+        self.layout().setColumnStretch(2, 50)
 
         starting = ", ".join(item_locations[2])
         starting_label = QtWidgets.QLabel(f"Starting items: {starting}")
         starting_label.setWordWrap(True)
-        self.layout.addWidget(starting_label, 0, 0, 1, 3)
+        self.layout().addWidget(starting_label, 0, 0, 1, 3)
 
         row_pos = 1
         if unsupported_game:
@@ -61,7 +67,7 @@ class GameLayout(QtWidgets.QWidget):
             )
 
             separator = QtWidgets.QLabel("")
-            self.layout.addWidget(separator, row_pos, 0, 1, 3)
+            self.layout().addWidget(separator, row_pos, 0, 1, 3)
             row_pos += 1
 
         self.build_items_display(
@@ -81,19 +87,19 @@ class GameLayout(QtWidgets.QWidget):
                 text.setStyleSheet(
                     f"border:1px solid black;background:{style.victory_background};color:black;"
                 )
-            text.setAlignment(QtCore.Qt.AlignCenter)
-            self.layout.addWidget(text, offset, 0, len(locations[item]), 1)
+            text.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
+            self.layout().addWidget(text, offset, 0, len(locations[item]), 1)
 
             for region, area, vanilla_item in locations[item]:
                 region_label = QtWidgets.QLabel(region)
                 region_label.setStyleSheet(
                     f"background:{style.background.get(region, style.fallback_background)};color:{style.foreground.get(region, style.fallback_foreground)};"
                 )
-                self.layout.addWidget(region_label, offset, 1, 1, 1)
+                self.layout().addWidget(region_label, offset, 1, 1, 1)
                 area_label = QtWidgets.QLabel(" ".join([area, vanilla_item]))
                 area_label.setStyleSheet(
                     f"background:{style.background.get(region, style.fallback_background)};color:{style.foreground.get(region, style.fallback_foreground)};"
                 )
-                self.layout.addWidget(area_label, offset, 2, 1, 1)
+                self.layout().addWidget(area_label, offset, 2, 1, 1)
                 offset = offset + 1
         return offset

--- a/src/gui/game_layout.py
+++ b/src/gui/game_layout.py
@@ -1,4 +1,3 @@
-from typing import Callable
 from PySide6 import QtCore, QtWidgets
 from world import World
 from gui.notification_dialog import NotificationDialog
@@ -14,8 +13,7 @@ from gui.layouts.fusion_style import FusionStyle
 class GameLayout(QtWidgets.QWidget):
     game_style: BaseLayout
     world: World
-    # We override this with the layout we intend to use
-    layout: Callable[..., QtWidgets.QGridLayout]
+    layout_obj: QtWidgets.QGridLayout
 
     def __init__(self, world: World):
         super().__init__()
@@ -41,15 +39,17 @@ class GameLayout(QtWidgets.QWidget):
             NotificationDialog.show(self, "Error", str(e))
             return
 
-        self.setLayout(QtWidgets.QGridLayout(self))
-        self.layout().setColumnStretch(0, 20)
-        self.layout().setColumnStretch(1, 30)
-        self.layout().setColumnStretch(2, 50)
+        self.layout_obj = QtWidgets.QGridLayout(self)
+        self.setLayout(self.layout_obj)
+
+        self.layout_obj.setColumnStretch(0, 20)
+        self.layout_obj.setColumnStretch(1, 30)
+        self.layout_obj.setColumnStretch(2, 50)
 
         starting = ", ".join(item_locations[2])
         starting_label = QtWidgets.QLabel(f"Starting items: {starting}")
         starting_label.setWordWrap(True)
-        self.layout().addWidget(starting_label, 0, 0, 1, 3)
+        self.layout_obj.addWidget(starting_label, 0, 0, 1, 3)
 
         row_pos = 1
         if unsupported_game:
@@ -67,7 +67,7 @@ class GameLayout(QtWidgets.QWidget):
             )
 
             separator = QtWidgets.QLabel("")
-            self.layout().addWidget(separator, row_pos, 0, 1, 3)
+            self.layout_obj.addWidget(separator, row_pos, 0, 1, 3)
             row_pos += 1
 
         self.build_items_display(
@@ -88,18 +88,18 @@ class GameLayout(QtWidgets.QWidget):
                     f"border:1px solid black;background:{style.victory_background};color:black;"
                 )
             text.setAlignment(QtCore.Qt.AlignmentFlag.AlignCenter)
-            self.layout().addWidget(text, offset, 0, len(locations[item]), 1)
+            self.layout_obj.addWidget(text, offset, 0, len(locations[item]), 1)
 
             for region, area, vanilla_item in locations[item]:
                 region_label = QtWidgets.QLabel(region)
                 region_label.setStyleSheet(
                     f"background:{style.background.get(region, style.fallback_background)};color:{style.foreground.get(region, style.fallback_foreground)};"
                 )
-                self.layout().addWidget(region_label, offset, 1, 1, 1)
+                self.layout_obj.addWidget(region_label, offset, 1, 1, 1)
                 area_label = QtWidgets.QLabel(" ".join([area, vanilla_item]))
                 area_label.setStyleSheet(
                     f"background:{style.background.get(region, style.fallback_background)};color:{style.foreground.get(region, style.fallback_foreground)};"
                 )
-                self.layout().addWidget(area_label, offset, 2, 1, 1)
+                self.layout_obj.addWidget(area_label, offset, 2, 1, 1)
                 offset = offset + 1
         return offset

--- a/src/gui/layouts/base_layout.py
+++ b/src/gui/layouts/base_layout.py
@@ -1,4 +1,10 @@
 class BaseLayout:
+    background: dict[str, str]
+    foreground: dict[str, str]
+    fallback_background: str
+    fallback_foreground: str
+    victory_background: str
+
     def __init__(self):
         self.background = {}
         self.foreground = {}

--- a/src/gui/main_window.py
+++ b/src/gui/main_window.py
@@ -7,13 +7,19 @@ import platform
 
 
 class MainWindow(QtWidgets.QMainWindow):
+    settings: Settings
+    options: dict
+    dark_mode: bool
+    text_size: int
+    scroll_area: QtWidgets.QScrollArea
+
     def __init__(self, file: str | None = None):
         super().__init__()
 
         self.settings = Settings()
         self.options = self.settings.get_options()
-        self.dark_mode: bool = self.options["dark_mode"]
-        self.text_size: int = self.options["text_size"]
+        self.dark_mode = self.options["dark_mode"]
+        self.text_size = self.options["text_size"]
 
         # Validate text size in case it was modified externally
         if self.text_size < 10 or self.text_size > 24:
@@ -24,8 +30,12 @@ class MainWindow(QtWidgets.QMainWindow):
 
         self.scroll_area = QtWidgets.QScrollArea()
         self.scroll_area.setGeometry(QtCore.QRect(0, 0, 800, 600))
-        self.scroll_area.setVerticalScrollBarPolicy(QtCore.Qt.ScrollBarAsNeeded)
-        self.scroll_area.setHorizontalScrollBarPolicy(QtCore.Qt.ScrollBarAsNeeded)
+        self.scroll_area.setVerticalScrollBarPolicy(
+            QtCore.Qt.ScrollBarPolicy.ScrollBarAsNeeded
+        )
+        self.scroll_area.setHorizontalScrollBarPolicy(
+            QtCore.Qt.ScrollBarPolicy.ScrollBarAsNeeded
+        )
         self.scroll_area.setWidgetResizable(True)
         self.scroll_area.setObjectName("scrollArea")
         self.scroll_area.setEnabled(True)
@@ -169,16 +179,16 @@ class MainWindow(QtWidgets.QMainWindow):
 
     # Override
     def dragMoveEvent(self, event: QtGui.QDragMoveEvent):
-        if event.mimeData().hasUrls:
-            event.setDropAction(QtCore.Qt.CopyAction)
+        if event.mimeData().hasUrls():
+            event.setDropAction(QtCore.Qt.DropAction.CopyAction)
             event.accept()
             return
         event.ignore()
 
     # Override
     def dropEvent(self, event: QtGui.QDropEvent):
-        if event.mimeData().hasUrls:
-            event.setDropAction(QtCore.Qt.CopyAction)
+        if event.mimeData().hasUrls():
+            event.setDropAction(QtCore.Qt.DropAction.CopyAction)
             event.accept()
 
             # Remove the URI stuff to convert into a path

--- a/src/gui/notification_dialog.py
+++ b/src/gui/notification_dialog.py
@@ -14,7 +14,7 @@ class NotificationDialog:
         message_label = QtWidgets.QLabel(message)
         dialog_layout.addWidget(message_label)
 
-        button_values = QtWidgets.QDialogButtonBox.Close
+        button_values = QtWidgets.QDialogButtonBox.StandardButton.Close
         button_box = QtWidgets.QDialogButtonBox(button_values)
         button_box.rejected.connect(dialog.reject)
         dialog_layout.addWidget(button_box)

--- a/src/settings.py
+++ b/src/settings.py
@@ -2,17 +2,23 @@ import json
 import os
 import platformdirs
 from pathlib import Path
+from typing import Any
 
 
 # TODO: Define an schema with default values
 class Settings:
+    settings_dir: Path
+    settings_file_path: Path
+    full_path: Path
+    options: dict
+
     def __init__(self):
-        self.settings_dir: Path = Path(
+        self.settings_dir = Path(
             platformdirs.user_config_dir(appname="RDVSpoilerLogParser")
         )
-        self.settings_file_path: Path = Path("rdvslp-settings.json")
-        self.full_path: Path = None
-        self.options: dict = dict()
+        self.settings_file_path = Path("rdvslp-settings.json")
+        self.full_path = None
+        self.options = {}
 
         self.full_path = self.settings_dir.joinpath(self.settings_file_path)
 
@@ -36,7 +42,7 @@ class Settings:
     def get_options(self) -> dict:
         return self.options
 
-    def write_option(self, option: str, value: str):
+    def write_option(self, option: str, value: Any):
         self.options[option] = value
 
     def save_options_to_file(self):

--- a/src/spoiler_file.py
+++ b/src/spoiler_file.py
@@ -5,9 +5,12 @@ from enum import Enum
 
 
 class SpoilerFile:
+    json: dict
+    world_names: list | None
+
     def __init__(self):
-        self.json: dict = None
-        self.world_names: list | None = None
+        self.json = {}
+        self.world_names = None
 
     def read(self, filename) -> SpoilerStatusEnum:
         try:

